### PR TITLE
TE5-T447 Explicit, Implicit and Pending Intent 

### PR DESCRIPTION
**Pull request by vishal2001prajapati**
…cyclerViewAdapter Complete.

### CODE REVIEW

File: app/src/main/java/com/example/myapplication3/Intent/FirstIntentActivity.kt [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+import com.example.myapplication3.ZERO
+
+class FirstIntentActivity : AppCompatActivity() {
+
+    lateinit var btnExplicit: Button
+    lateinit var editTextData: EditText
+    lateinit var btnImplicit: Button
+    lateinit var btnPendingIntent: Button
+    lateinit var notificationManager: NotificationManager
+    private val channelId = "i.app.noffification"
```
Comment (purvesh-d) [inline app/src/main/java/com/example/myapplication3/Intent/FirstIntentActivity.kt]: move to constatnt

File: app/src/main/java/com/example/myapplication3/Intent/FirstIntentActivity.kt [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    private val notificationId = 101
```
Comment (purvesh-d) [inline app/src/main/java/com/example/myapplication3/Intent/FirstIntentActivity.kt]: same

File: app/src/main/java/com/example/myapplication3/Intent/FirstIntentActivity.kt (hunk 1 above)
Comment (Purvesh-Dodiya) [inline app/src/main/java/com/example/myapplication3/Intent/FirstIntentActivity.kt]: move to constant file

File: app/src/main/java/com/example/myapplication3/Intent/FirstIntentActivity.kt [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+                    description = descriptionText
+                }
+                val notificationManager: NotificationManager = getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
+                notificationManager.createNotificationChannel(channel)
+            }
+            val pendingIntent = PendingIntent.getActivity(this, ZERO, intent, PendingIntent.FLAG_IMMUTABLE)
+            val bitmapLargeIcon = BitmapFactory.decodeResource(applicationContext.resources, R.drawable.bookmark)
+            val builder = NotificationCompat.Builder(this, channelId)
+                    .setSmallIcon(R.drawable.bookmark)
+                    .setContentTitle("Example Title")
```
Comment (Purvesh-Dodiya) [inline app/src/main/java/com/example/myapplication3/Intent/FirstIntentActivity.kt]: remove static string

**Review by Purvesh-Dodiya — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/myapplication3/Intent/FirstIntentActivity.kt [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+class FirstIntentActivity : AppCompatActivity() {
+
+    lateinit var btnExplicit: Button
+    lateinit var editTextData: EditText
+    lateinit var btnImplicit: Button
+    lateinit var btnPendingIntent: Button
+    lateinit var notificationManager: NotificationManager
+
+    private val notificationId = 101
```
Comment (purvesh-d) [inline app/src/main/java/com/example/myapplication3/Intent/FirstIntentActivity.kt]: pending

File: app/src/main/java/com/example/myapplication3/Intent/FirstIntentActivity.kt [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+                    .setContentIntent(pendingIntent)
+                    .setPriority(NotificationCompat.PRIORITY_HIGH)
+            with(NotificationManagerCompat.from(this)) {
+                notify(notificationId, builder.build())
+            }
+        }
+        btnExplicit.setOnClickListener {
+            val data = editTextData.text.toString()
+            val intent = Intent(this, SecondIntentActivity::class.java)
+            intent.putExtra("Data", data)
```
Comment (purvesh-d) [inline app/src/main/java/com/example/myapplication3/Intent/FirstIntentActivity.kt]: move key to constant

File: app/src/main/java/com/example/myapplication3/Intent/FirstIntentActivity.kt [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+            startActivity(intent)
+        }
+        btnImplicit.setOnClickListener {
+            val intent = Intent(Intent.ACTION_VIEW, Uri.parse("https://www.youtube.com/"))
```
Comment (purvesh-d) [inline app/src/main/java/com/example/myapplication3/Intent/FirstIntentActivity.kt]: static string

File: app/src/main/java/com/example/myapplication3/Intent/SecondIntentActivity.kt [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+import com.example.myapplication3.R
+
+class SecondIntentActivity : AppCompatActivity() {
+
+    lateinit var textView: TextView
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_second_intent)
+        val userData = intent.getStringExtra("Data")
```
Comment (purvesh-d) [inline app/src/main/java/com/example/myapplication3/Intent/SecondIntentActivity.kt]: static key

**Review by purvesh-d — CHANGES_REQUESTED** [review]

**Review by purvesh-d — APPROVED** [review]

**Pull request merged**